calypso/service.go

```diff
@@ -434,6 +449,24 @@ func (s *Service) ReshareLTS(req *ReshareLTS) (*ReshareLTSReply, error) {
 	return &ReshareLTSReply{}, nil
 }
 
+// Private service endpoint that extends the set of valid peers with the ones
+// in the provided roster.
+func (s *Service) updateValidPeers(req *updateValidPeers) (
+	*updateValidPeersReply, error) {
+	currentRoster := s.storage.Rosters[req.ByzcoinID]
+
+	// Build the new set as the union of the current and new rosters
+	newPeerSet := []*network.ServerIdentity{}
+	if currentRoster != nil {
+		newPeerSet = append(newPeerSet, currentRoster.List...)
+	}
+	newPeerSet = append(newPeerSet, req.NewRoster.List...)
+
+	s.SetValidPeers(s.NewPeerSetID(req.ByzcoinID[:]), newPeerSet)
```

Comment:
@ineiti Is that ID a unique identifier for the LTS?

Reply:
TLDR: no, it's not the unique identifier

Longer version:

The LTS stores two things:
- the `ByzCoinID`, which is the hash of the genesis-block, to know how to verify proofs
- a list of `LTSIDs` that have been stored in the byzcoin-instance with the `ByzCoinID`

Every time a client wants to add an LTSID or wants to do a re-encryption, he has to proof he has the right by providing a proof with the `ByzCoinID` as the base.

So the valid peers for Calypso is not the byzcoin-roster, but the roster of the LTS, which can (and are) different. You can find this information in the `s.storage.Rosters` map.

Reply:
And in fact if you use `ByzCoinID` here, it will overwrite the set of peers for different LTSIDs. So you'll have to change for the LTSID.

Reply:
Summary after discussion: it is actually the correct unique ID, just incorrectly named (it should be named `LTSID`) :-)